### PR TITLE
feat(ci): automatic creation of test plans and naming convention

### PR DESCRIPTION
**Pull request by a-eljazouly**
## Description
Automate the creation of TP in NIGHTLY

MON-24083

**Fixes** # (issue)

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 21.10.x
- [ ] 22.04.x
- [ ] 22.10.x
- [ ] 23.04.x
- [x] 23.10.x (master)

#### Community contributors & Centreon team

- [x] I have followed the **coding style guidelines** provided by Centreon
- [x] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [x] I have commented my code, especially **hard-to-understand areas** of the PR.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: .github/actions/get-test-plan-key/action.yml [hunk 1]
```diff
@@ -0,0 +1,75 @@
+name: get-test-plan-key
+description: "Determine the TP key if the TP exist, create it if doesn't"
```
Comment (sc979) [inline .github/actions/get-test-plan-key/action.yml:2]:
```suggestion
description: "Determine the TP key if the TP exist, create it if it doesn't"
```

File: .github/actions/get-test-plan-key/action.yml [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+runs:
+  using: "composite"
+  steps:
+    - name: Get the TestPlanKey
+      id: get_test_plan_key
+      run: |
+        # Use a GraphQL query to get all test plans matching the JQL
+        graphql_query='{
+          "query":"query GetTestPlans($jql: String, $limit: Int!) { getTestPlans(jql: $jql, limit: $limit) { total results { issueId jira(fields: [\"summary\", \"key\"]) } } }",
+          "variables":{"jql": "project = MON","limit": 100}}'
```
Comment (sc979) [inline .github/actions/get-test-plan-key/action.yml:25]:
```suggestion
          "variables":{"jql": "project = MON","limit": $limit}}'
```

File: .github/actions/get-test-plan-key/action.yml [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+        let test_plan_key = '';
+        data.forEach(item => {
+            const key = item.jira.key;
+            const summary = item.jira.summary;
+            if (summary === "NIGHTLY OSS ${{ inputs.major_version }}") {
+              test_plan_key = key;
+            }
+        });
+        console.log(test_plan_key);
+        EOF)
```
Comment (sc979) [inline .github/actions/get-test-plan-key/action.yml:43]: Couldn't you use bash instead ?

Comment (a-eljazouly) [reply .github/actions/get-test-plan-key/action.yml:43]: @sc979 unfortunately i couldn't :( , i tried to use jq command but the json result is not well formatted

File: .github/actions/get-test-plan-key/action.yml (hunk 2 above)
Comment (a-eljazouly) [reply .github/actions/get-test-plan-key/action.yml:25]: @sc979, it won't work because `$limit` is a variable that hasn't been set.

Comment (sc979) [reply .github/actions/get-test-plan-key/action.yml:25]: Ah indeed I read it wrong

File: .github/actions/get-test-plan-key/action.yml (hunk 3 above)
Comment (kduret) [reply .github/actions/get-test-plan-key/action.yml:43]: can you provide us an example of `.data.getTestPlans.results` please ?
to check with `jq` outside of the pipeline

Comment (a-eljazouly) [reply .github/actions/get-test-plan-key/action.yml:43]: @kduret yes ofc :
https://github.com/centreon/centreon/actions/runs/6639197899/job/18037936008
https://github.com/centreon/centreon/actions/runs/6639595838/job/18039214707
https://github.com/centreon/centreon/actions/runs/6640036705/job/18040728876

Comment (kduret) [reply .github/actions/get-test-plan-key/action.yml:43]: indeed it seems there are some special characters, but it works with : 
```
DATA='[{"issueId":"1","jira":{"key":"MON-1","summary":"Test plan 1"}},{"issueId":"2","jira":{"key":"MON-2","summary":"test plan 2"}}]'
echo $DATA | jq -c '.[].jira' | while read row; do
  echo "${row}" | jq '.key'
done
```

File: .github/workflows/cypress-e2e-parallelization.yml [hunk 4]
```diff
@@ -154,9 +157,10 @@ jobs:
           path: ${{ inputs.module_name }}/tests/e2e/cypress/cucumber-logs/*.json
           retention-days: 1
 
+  # This action is only for NIGHTLY tests working only for unstable and testing branches
   import-reports-to-xray:
     needs: [cypress-e2e-test-run]
-    if: ${{ inputs.stability != 'canary' }}
+    if: ${{ inputs.stability != 'canary' && github.event_name == 'schedule' }}
```
Comment (kduret) [inline .github/workflows/cypress-e2e-parallelization.yml:163]:
```suggestion
    if: ${{ inputs.stability == 'testing' || github.event_name == 'schedule' }}
```

File: .github/actions/get-test-plan-key/action.yml [hunk 5]
```diff
@@ -27,20 +27,30 @@ runs:
...
+        # let test_plan_key = '';
+        # data.forEach(item => {
+        #     const key = item.jira.key;
+        #     const summary = item.jira.summary;
+        #     if (summary === "NIGHTLY OSS ${{ inputs.major_version }}") {
+        #       test_plan_key = key;
+        #     }
+        # });
+        # console.log(test_plan_key);
+        # EOF)
```
Comment (sc979) [inline .github/actions/get-test-plan-key/action.yml:43]: obsolete lines should be removed. you will get them in the git log history if needed

File: .github/workflows/cypress-e2e-parallelization.yml [hunk 6]
```diff
@@ -154,9 +157,10 @@ jobs:
           path: ${{ inputs.module_name }}/tests/e2e/cypress/cucumber-logs/*.json
           retention-days: 1
 
+  # This action is only for NIGHTLY and testing branches
```
Comment (kduret) [inline .github/workflows/cypress-e2e-parallelization.yml:160]:
```suggestion
  # This action is only for NIGHTLY, unstable and testing branches
```

File: .github/workflows/cypress-e2e-parallelization.yml [hunk 7]
```diff
@@ -154,9 +157,10 @@ jobs:
...
   import-reports-to-xray:
     needs: [cypress-e2e-test-run]
-    if: ${{ inputs.stability != 'canary' }}
+    if: always()
```
Comment (kduret) [inline .github/workflows/cypress-e2e-parallelization.yml:163]:
```suggestion
    if: ${{ contains(fromJson('["testing", "unstable"]'), inputs.stability) }}
```

File: .github/workflows/cypress-e2e-parallelization.yml [hunk 8]
```diff
@@ -281,7 +232,7 @@ jobs:
           jira_payload='
           {
             "fields": {
-              "summary": "${{ github.run_id}} E2E Tests",
+              "summary": "RT OSS ${{ inputs.major_version }}.${{ inputs.minor_version }} ",
```
Comment (kduret) [inline .github/workflows/cypress-e2e-parallelization.yml:235]:
```suggestion
              "summary": "${{format('{0} OSS, inputs.major_version{1}', github.event_name == 'schedule' && 'NIGHTLY' || 'RT', inputs.stability == 'testing' && format('.{0}', inputs.minor_version) || '')}} ",
```

**Review by kduret — DISMISSED** [review]

**Review by sc979 — DISMISSED** [review]

**Review by kduret — APPROVED** [review]

**Review by sc979 — APPROVED** [review]

**Pull request merged**